engine/runtime-integration-tests/src/test/java/org/enso/compiler/test/IfThenElseTest.java

```diff
@@ -96,6 +96,43 @@ public void variableDefinedInElse() throws Exception {
     assertEquals("Bad:False", check.execute(false).asString());
   }
 
+  @Test
+  public void variableInMultipleIfBranches() throws Exception {
+    var code =
+        """
+    check x =
+        if x then
+            xt = "Yes"
+        if x.not then
+            xt = "No"
+        "Hooo"
+    """;
+    var check = ContextUtils.getMethodFromModule(ctx, code, "check");
+
+    assertEquals("Hooo", check.execute(true).asString());
+    assertEquals("Hooo", check.execute(false).asString());
+  }
+
+  @Test
+  public void variableNotVisibleAfterBranches() throws Exception {
```

Comment:
Testing scopes of if then else is a good idea. I would like to see more expanded tests. Something like:
```
x = True
if x then
  y = True
  if x then if y then "Foo"
```
and
```
x = False
cond = False
if cond then "cond" else
  y = False
  if x then "x" else
    if y then "y" else 42
```
that is, try to use bindings from various parent scopes in the condition, true branch and false branch.

These tests could be placed in `Base_Tests` as well I believe.